### PR TITLE
[Mobile] - Update E2E config to support an iPad simulator

### PR DESCRIPTION
**Pull request by geriux**
**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5756

## What?
This PR adds support for iPad E2E testing on Gutenberg mobile, on this repo iPad tests are not set up.

## Why?
While working on refactoring the BlockList we realized there are quite a few cases with block alignments and complex blocks like Columns, and Group, where different screen sizes have different expected behaviors.

## How?

- It updates `iosLocal` and `iosServer` objects into functions to support params. In this case, a new object param is set to be able to pass options like: `iPadDevice`.
- Within the test setup code, it will check if an `IPAD` environment variable is set and it will pass this value to the server config.

## Testing Instructions
N/A

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

File: packages/react-native-editor/__device-tests__/helpers/utils.js [hunk 1]
```diff
@@ -115,10 +116,10 @@ const setupDriver = async () => {
 			desiredCaps.app = `sauce-storage:Gutenberg-${ safeBranchName }.apk`; // App should be preloaded to sauce storage, this can also be a URL.
 		}
 	} else {
-		desiredCaps = { ...iosServer };
+		desiredCaps = { ...iosServer( { iPadDevice } ) };
```
Comment (dcalhoun) [inline packages/react-native-editor/__device-tests__/helpers/utils.js]: I may be overlooking something, but the object spread on this line and its sibling seem unnecessary. Feel free to ignore if you disagree.

```suggestion
		desiredCaps = iosServer( { iPadDevice } );
```

File: packages/react-native-editor/__device-tests__/helpers/utils.js [hunk 2]
```diff
@@ -115,10 +116,10 @@ const setupDriver = async () => {
...
 		desiredCaps.app = `sauce-storage:Gutenberg-${ safeBranchName }.app.zip`; // App should be preloaded to sauce storage, this can also be a URL.
 		if ( isLocalEnvironment() ) {
-			desiredCaps = { ...iosLocal };
+			desiredCaps = { ...iosLocal( { iPadDevice } ) };
```
Comment (dcalhoun) [inline packages/react-native-editor/__device-tests__/helpers/utils.js]: I may be overlooking something, but the object spread on this line and its sibling seem unnecessary.

```suggestion
			desiredCaps = iosLocal( { iPadDevice } );
```

**Review by dcalhoun — APPROVED** [review]

File: packages/react-native-editor/__device-tests__/helpers/utils.js (hunk 1 above)
Comment (geriux) [reply packages/react-native-editor/__device-tests__/helpers/utils.js]: Nice catch! I updated in https://github.com/WordPress/gutenberg/pull/50528/commits/535548f267b1caeedea276f6400944033526cc87

File: packages/react-native-editor/__device-tests__/helpers/utils.js (hunk 2 above)
Comment (geriux) [reply packages/react-native-editor/__device-tests__/helpers/utils.js]: Updated in https://github.com/WordPress/gutenberg/pull/50528/commits/535548f267b1caeedea276f6400944033526cc87

**Pull request merged**